File: AnalyticsAppObjC/AnalyticsAppObjC/CustomLogger/CustomLogger.m
//
//  CustomLogger.m
//  AnalyticsAppObjC
//
//  Created by Satheesh Kannan on 15/05/25.
//

#import "CustomLogger.h"

@implementation CustomLogger

- (void)verbose:(NSString * _Nonnull)log {
    NSLog(@"[Analytics-ObjC] :: Verbose :: %@", log);
}

- (void)debug:(NSString * _Nonnull)log {
    NSLog(@"[Analytics-ObjC] :: Debug :: %@", log);
}

- (void)info:(NSString * _Nonnull)log {
    NSLog(@"[Analytics-ObjC] :: Info :: %@", log);
}

- (void)warn:(NSString * _Nonnull)log {
    NSLog(@"[Analytics-ObjC] :: Warn :: %@", log);
}

- (void)errorLog:(NSString * _Nonnull)log error:(NSError * _Nullable)error {
    NSLog(@"[Analytics-ObjC] :: Error :: %@", log);
    if (error != nil) {
        NSLog(@"[Analytics-ObjC] :: Error Details :: %@", [error debugDescription]);
    }
}

@end

File: AnalyticsApp/AnalyticsApp/CustomLogger.swift
//
//  CustomLogger.swift
//  AnalyticsApp
//
//  Created by Satheesh Kannan on 22/04/25.
//

import Foundation
import Analytics

class CustomLogger: Logger {
    func verbose(log: String) {
        print("[Analytics-Swift] :: Verbose :: \(log)")
    }
    
    func debug(log: String) {
        print("[Analytics-Swift] :: Debug :: \(log)")
    }
    
    func info(log: String) {
        print("[Analytics-Swift] :: Info :: \(log)")
    }
    
    func warn(log: String) {
        print("[Analytics-Swift] :: Warn :: \(log)")
    }
    
    func error(log: String, error: (any Error)?) {
        print("[Analytics-Swift] :: Error :: \(log)")
        if let error {
            print("[Analytics-Swift] :: Error Details :: \(error)")
        }
    }
}
